examples/op/l1-to-l2/methods/guest/src/main.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright 2024 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![allow(unused_doc_comments)]
+#![no_main]
+
+use l1_to_l2_core::{CALL, CALLER, CONTRACT};
+use risc0_steel::{
+    ethereum::{EthEvmInput, ETH_MAINNET_CHAIN_SPEC},
+    Contract,
+};
+use risc0_zkvm::guest::env;
+
+risc0_zkvm::guest::entry!(main);
+
+fn main() {
+    // Read the input from the guest environment.
+    let input: EthEvmInput = env::read();
+
+    // Create the environment and add the commitment.
+    let env = input.into_env().with_chain_spec(&ETH_MAINNET_CHAIN_SPEC);
+    env::commit_slice(&env.commitment().abi_encode());
+
+    // Execute the call.
+    let contract = Contract::new(CONTRACT, &env);
+    let returns = contract.call_builder(&CALL).from(CALLER).call();
+
+    // Commit the result.
+    env::commit(&returns._0)
```

Comment:
I think it'd be a better pattern to use something like encoding a tuple or another lightweight data structuring to combine the data and commit with one call, rather than calling commit twice. Concatenating data by calling commit more than once seems more error prone, potentially resulting in confusing decoding issues (and potentially security bugs if it ends up producing a byte-string with multiple valid decodings).
```suggestion
    env::commit_slice((&env.commitment(), &returns._0).abi_encode())
```